File: suc/swift/094.swift
import SwiftUI

@main
struct SecureKeyApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

struct ContentView: View {
    var body: some View {
        SecureKeyScreen()
            .preferredColorScheme(.light)
            .statusBarHidden(true)
    }
}

// MARK: - Models
struct AccessPoint: Identifiable {
    let id: String
    let name: String
    let status: String
    let isLocked: Bool
    let lastAccessed: String
}

// MARK: - Colors
struct SecureColors {
    static let primary = Color.black
    static let surface = Color(red: 0.97, green: 0.97, blue: 0.97) // #F8F8F8
    static let locked = Color(red: 0.13, green: 0.13, blue: 0.13) // #212121
    static let unlocked = Color(red: 0.30, green: 0.69, blue: 0.31) // #4CAF50
    static let border = Color(red: 0.94, green: 0.94, blue: 0.94) // #EFEFEF
    static let textGray = Color.gray
    static let backgroundWhite = Color.white
}

// MARK: - Main Screen
struct SecureKeyScreen: View {
    @State private var masterLockState = true
    
    let accessPoints = [
        AccessPoint(id: "1", name: "Main Entrance", status: "Active", isLocked: true, lastAccessed: "2h ago"),
        AccessPoint(id: "2", name: "Garage Door", status: "Active", isLocked: false, lastAccessed: "12m ago"),
        AccessPoint(id: "3", name: "Server Room", status: "Restricted", isLocked: true, lastAccessed: "Yesterday")
    ]
    
    var body: some View {
        ZStack {
            SecureColors.backgroundWhite.ignoresSafeArea()
            
            ScrollView(showsIndicators: false) {
                VStack(alignment: .leading, spacing: 0) {
                    
                    // Header
                    HeaderSection()
                        .padding(.top, 24)
                    
                    Spacer().frame(height: 40)
                    
                    // Master Status Card
                    MasterStatusCard(isLocked: masterLockState) {
                        masterLockState.toggle()
                    }
                    
                    Spacer().frame(height: 32)
                    
                    // Access Points Header
                    Text("My Access Points")
                        .font(.system(size: 16, weight: .bold)) // TitleMedium approx
                        .foregroundColor(Color(white: 0.2)) // DarkGray
                    
                    Spacer().frame(height: 16)
                    
                    // Access Points List
                    VStack(spacing: 12) {
                        ForEach(accessPoints) { point in
                            AccessPointRow(item: point)
                        }
                    }
                    
                    Spacer().frame(height: 32)
                    
                    // Recent Activity Header
                    Text("Recent Activity")
                        .font(.system(size: 16, weight: .bold)) // TitleMedium approx
                        .foregroundColor(Color(white: 0.2))
                    
                    Spacer().frame(height: 16)
                    
                    // Activity Log
                    VStack(spacing: 0) {
                        ActivityLogItem(time: "10:42 AM", event: "Main Gate unlocked", user: "You")
                        ActivityLogItem(time: "09:15 AM", event: "Guest Entry Denied", user: "System")
                        ActivityLogItem(time: "08:30 AM", event: "Garage Door opened", user: "Alice")
                    }
                    
                    Spacer().frame(height: 24)
                }
                .padding(.horizontal, 24)
            }
        }
    }
}

// MARK: - Components

struct HeaderSection: View {
    var body: some View {
        HStack(alignment: .center) {
            VStack(alignment: .leading, spacing: 2) {
                Text("Welcome back,")
                    .font(.system(size: 14)) // BodyMedium approx
                    .foregroundColor(SecureColors.textGray)
                
                Text("David Miller")
                    .font(.system(size: 24, weight: .black)) // HeadlineSmall + ExtraBold approx
                    .foregroundColor(.black)
            }
            
            Spacer()
            
            ZStack {
                Circle()
                    .fill(Color(red: 0.96, green: 0.96, blue: 0.96)) // #F5F5F5
                    .frame(width: 48, height: 48)
                    .overlay(
                        Circle()
                            .stroke(Color(red: 0.88, green: 0.88, blue: 0.88), lineWidth: 1) // #E0E0E0
                    )
                
                Image(systemName: "person.fill")
                    .font(.system(size: 20))
                    .foregroundColor(.black)
            }
        }
    }
}

struct MasterStatusCard: View {
    let isLocked: Bool
    let onToggle: () -> Void
    
    var statusColor: Color {
        isLocked ? SecureColors.locked : SecureColors.unlocked
    }
    
    var body: some View {
        VStack(spacing: 0) {
            // Icon Button
            Button(action: onToggle) {
                ZStack {
                    Circle()
                        .fill(Color.white)
                        .frame(width: 80, height: 80)
                        .overlay(
                            Circle()
                                .stroke(statusColor.opacity(0.1), lineWidth: 2)
                        )
                    
                    Image(systemName: isLocked ? "lock.fill" : "checkmark.circle.fill")
                        .font(.system(size: 36))
                        .foregroundColor(statusColor)
                }
            }
            .padding(.bottom, 16)
            
            // Status Text
            Text(isLocked ? "SECURE" : "UNLOCKED")
                .font(.system(size: 22, weight: .black))
                .tracking(2)
                .foregroundColor(statusColor)
            
            Spacer().frame(height: 8)
            
            // Subtext
            Text(isLocked ? "All perimeter doors are locked." : "Perimeter access is currently open.")
                .font(.system(size: 14))
                .foregroundColor(SecureColors.textGray)
        }
        .padding(24)
        .frame(maxWidth: .infinity)
        .background(SecureColors.surface)
        .cornerRadius(24)
        .overlay(
            RoundedRectangle(cornerRadius: 24)
                .stroke(SecureColors.border, lineWidth: 1)
        )
    }
}

struct AccessPointRow: View {
    let item: AccessPoint
    @State private var checked: Bool
    
    init(item: AccessPoint) {
        self.item = item
        _checked = State(initialValue: !item.isLocked)
    }
    
    var body: some View {
        HStack {
            HStack(spacing: 16) {
                ZStack {
                    RoundedRectangle(cornerRadius: 10)
                        .fill(Color(red: 0.98, green: 0.98, blue: 0.98)) // #FAFAFA
                        .frame(width: 40, height: 40)
                    
                    Image(systemName: "house.fill")
                        .font(.system(size: 20))
                        .foregroundColor(.black)
                }
                
                VStack(alignment: .leading, spacing: 2) {
                    Text(item.name)
                        .font(.system(size: 16, weight: .semibold)) // BodyLarge + SemiBold
                        .foregroundColor(.black)
                    
                    Text(item.lastAccessed)
                        .font(.system(size: 11)) // LabelSmall
                        .foregroundColor(Color(white: 0.8)) // LightGray approximation
                }
            }
            
            Spacer()
            
            Toggle("", isOn: $checked)
                .labelsHidden()
                .toggleStyle(SwitchToggleStyle(tint: .black))
        }
        .padding(16)
        .background(Color.white)
        .cornerRadius(16)
        .overlay(
            RoundedRectangle(cornerRadius: 16)
                .stroke(Color(red: 0.93, green: 0.93, blue: 0.93), lineWidth: 1) // #EEEEEE
        )
    }
}

struct ActivityLogItem: View {
    let time: String
    let event: String
    let user: String
    
    var body: some View {
        HStack(alignment: .center, spacing: 0) {
            Text(time)
                .font(.system(size: 12)) // LabelMedium
                .foregroundColor(SecureColors.textGray)
                .frame(width: 70, alignment: .leading)
            
            Rectangle()
                .fill(Color(red: 0.93, green: 0.93, blue: 0.93)) // #EEEEEE
                .frame(width: 1, height: 20)
                .padding(.trailing, 16)
            
            VStack(alignment: .leading, spacing: 2) {
                Text(event)
                    .font(.system(size: 14, weight: .medium)) // BodyMedium + Medium
                    .foregroundColor(.black)
                
                Text("by \(user)")
                    .font(.system(size: 11)) // LabelSmall
                    .foregroundColor(SecureColors.textGray)
            }
            
            Spacer()
        }
        .padding(.vertical, 12)
    }
}

File: suc/kotlin/094.kt
package com.example.myapplication

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.DateRange
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Divider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.SideEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Configuration for Immersive Mode (Hide System Bars)
        val windowInsetsController = WindowCompat.getInsetsController(window, window.decorView)
        windowInsetsController.systemBarsBehavior =
            WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE

        // Hide both status bar and navigation bar
        windowInsetsController.hide(WindowInsetsCompat.Type.systemBars())

        setContent {
            // Re-apply immersive mode on recomposition ensures it stays hidden
            SideEffect {
                windowInsetsController.hide(WindowInsetsCompat.Type.systemBars())
            }

            MaterialTheme(
                colorScheme = MaterialTheme.colorScheme.copy(
                    background = Color.White,
                    surface = Color.White,
                    onBackground = Color.Black,
                    onSurface = Color.Black
                )
            ) {
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    color = Color.White
                ) {
                    SecureKeyScreen()
                }
            }
        }
    }
}

// --- Domain Models ---
data class AccessPoint(
    val id: String,
    val name: String,
    val status: String,
    val isLocked: Boolean,
    val lastAccessed: String
)

// --- Composable: Main Screen ---
@Composable
fun SecureKeyScreen() {
    val scrollState = rememberScrollState()

    // Mock State
    var masterLockState by remember { mutableStateOf(true) }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(24.dp)
            .verticalScroll(scrollState),
        verticalArrangement = Arrangement.Top
    ) {
        // Top Header
        HeaderSection()

        Spacer(modifier = Modifier.height(40.dp))

        // Master Control Status
        MasterStatusCard(
            isLocked = masterLockState,
            onToggle = { masterLockState = !masterLockState }
        )

        Spacer(modifier = Modifier.height(32.dp))

        // Quick Access List
        Text(
            text = "My Access Points",
            style = MaterialTheme.typography.titleMedium,
            fontWeight = FontWeight.Bold,
            color = Color.DarkGray
        )

        Spacer(modifier = Modifier.height(16.dp))

        AccessPointList()

        Spacer(modifier = Modifier.height(32.dp))

        // Recent Activity / Visitor Log
        Text(
            text = "Recent Activity",
            style = MaterialTheme.typography.titleMedium,
            fontWeight = FontWeight.Bold,
            color = Color.DarkGray
        )
        Spacer(modifier = Modifier.height(16.dp))

        ActivityLogItem(time = "10:42 AM", event = "Main Gate unlocked", user = "You")
        ActivityLogItem(time = "09:15 AM", event = "Guest Entry Denied", user = "System")
        ActivityLogItem(time = "08:30 AM", event = "Garage Door opened", user = "Alice")
    }
}

// --- Composable: Header ---
@Composable
fun HeaderSection() {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Column {
            Text(
                text = "Welcome back,",
                style = MaterialTheme.typography.bodyMedium,
                color = Color.Gray
            )
            Text(
                text = "David Miller",
                style = MaterialTheme.typography.headlineSmall,
                fontWeight = FontWeight.ExtraBold
            )
        }

        Box(
            modifier = Modifier
                .size(48.dp)
                .clip(CircleShape)
                .background(Color(0xFFF5F5F5))
                .border(1.dp, Color(0xFFE0E0E0), CircleShape),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = Icons.Default.Person,
                contentDescription = "Profile",
                tint = Color.Black
            )
        }
    }
}

// --- Composable: Master Status Card ---
@Composable
fun MasterStatusCard(isLocked: Boolean, onToggle: () -> Unit) {
    val statusColor = if (isLocked) Color(0xFF212121) else Color(0xFF4CAF50)
    val statusText = if (isLocked) "SECURE" else "UNLOCKED"
    val subText = if (isLocked) "All perimeter doors are locked." else "Perimeter access is currently open."

    Card(
        shape = RoundedCornerShape(24.dp),
        colors = CardDefaults.cardColors(containerColor = Color(0xFFF8F8F8)),
        elevation = CardDefaults.cardElevation(defaultElevation = 0.dp),
        modifier = Modifier
            .fillMaxWidth()
            .border(1.dp, Color(0xFFEFEFEF), RoundedCornerShape(24.dp))
    ) {
        Column(
            modifier = Modifier
                .padding(24.dp)
                .fillMaxWidth(),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Box(
                modifier = Modifier
                    .size(80.dp)
                    .clip(CircleShape)
                    .background(Color.White)
                    .clickable { onToggle() }
                    .border(2.dp, statusColor.copy(alpha = 0.1f), CircleShape),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = if (isLocked) Icons.Default.Lock else Icons.Default.CheckCircle,
                    contentDescription = "Status Icon",
                    tint = statusColor,
                    modifier = Modifier.size(36.dp)
                )
            }

            Spacer(modifier = Modifier.height(16.dp))

            Text(
                text = statusText,
                style = MaterialTheme.typography.titleLarge,
                fontWeight = FontWeight.Black,
                letterSpacing = 2.sp,
                color = statusColor
            )

            Spacer(modifier = Modifier.height(8.dp))

            Text(
                text = subText,
                style = MaterialTheme.typography.bodyMedium,
                color = Color.Gray
            )
        }
    }
}

// --- Composable: Access Points ---
@Composable
fun AccessPointList() {
    val accessPoints = listOf(
        AccessPoint("1", "Main Entrance", "Active", true, "2h ago"),
        AccessPoint("2", "Garage Door", "Active", false, "12m ago"),
        AccessPoint("3", "Server Room", "Restricted", true, "Yesterday")
    )

    Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
        accessPoints.forEach { point ->
            AccessPointRow(point)
        }
    }
}

@Composable
fun AccessPointRow(item: AccessPoint) {
    var checked by remember { mutableStateOf(!item.isLocked) }

    Row(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.White)
            .border(1.dp, Color(0xFFEEEEEE), RoundedCornerShape(16.dp))
            .padding(16.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Box(
                modifier = Modifier
                    .size(40.dp)
                    .clip(RoundedCornerShape(10.dp))
                    .background(Color(0xFFFAFAFA)),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = Icons.Default.Home, // Using generic Home icon for simplicity
                    contentDescription = null,
                    tint = Color.Black,
                    modifier = Modifier.size(20.dp)
                )
            }
            Spacer(modifier = Modifier.width(16.dp))
            Column {
                Text(
                    text = item.name,
                    style = MaterialTheme.typography.bodyLarge,
                    fontWeight = FontWeight.SemiBold
                )
                Text(
                    text = item.lastAccessed,
                    style = MaterialTheme.typography.labelSmall,
                    color = Color.LightGray
                )
            }
        }

        Switch(
            checked = checked,
            onCheckedChange = { checked = it },
            colors = SwitchDefaults.colors(
                checkedThumbColor = Color.White,
                checkedTrackColor = Color.Black,
                uncheckedThumbColor = Color.Gray,
                uncheckedTrackColor = Color(0xFFEEEEEE),
                uncheckedBorderColor = Color.Transparent
            )
        )
    }
}

// --- Composable: Activity Log ---
@Composable
fun ActivityLogItem(time: String, event: String, user: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(
            text = time,
            style = MaterialTheme.typography.labelMedium,
            color = Color.Gray,
            modifier = Modifier.width(70.dp)
        )

        Divider(
            modifier = Modifier
                .height(20.dp)
                .width(1.dp),
            color = Color(0xFFEEEEEE)
        )

        Spacer(modifier = Modifier.width(16.dp))

        Column {
            Text(
                text = event,
                style = MaterialTheme.typography.bodyMedium,
                fontWeight = FontWeight.Medium
            )
            Text(
                text = "by $user",
                style = MaterialTheme.typography.labelSmall,
                color = Color.Gray
            )
        }
    }
}

// --- Composable: Bottom Navigation (Simplified visual representation) ---
// Not used in main scroll, but could be added if layout required fixed bottom bar.
// For this single-screen flow, we keep it clean without a nav bar as requested.

@Preview(showBackground = true)
@Composable
fun PreviewSecureKey() {
    MaterialTheme(
        colorScheme = MaterialTheme.colorScheme.copy(
            background = Color.White,
            surface = Color.White
        )
    ) {
        SecureKeyScreen()
    }
}